base/src/main/resources/db/migration/V84__tenant_delete_cascade.sql

```diff
@@ -0,0 +1,182 @@
+    (SELECT conrelid::regclass::text AS table_name, conname, pg_get_constraintdef(oid) as constraintdef
+     FROM pg_constraint
+     WHERE contype = 'f');
+
+CREATE TEMP TABLE table_constraints_ AS (select *
+                                         from information_schema.table_constraints);
+CREATE TEMP TABLE key_column_usage_ AS (select *
+                                        from information_schema.key_column_usage);
+CREATE TEMP TABLE constraint_column_usage_ AS (select *
+                                               from information_schema.constraint_column_usage);
+
+CREATE TEMP TABLE filtered_sys_tenant_tables AS
+    (SELECT ccu.table_name     AS foreign_table_name,
+            ccu.column_name    as foreign_column_name,
+            tc.constraint_name,
+            kcu.table_name     as table_name,
+            ftfk.constraintdef as constraint_def,
+            kcu.column_name    as column_name
+     FROM table_constraints_ AS tc
+              JOIN key_column_usage_ AS kcu
+                   ON tc.constraint_name = kcu.constraint_name
+                       AND tc.table_schema = kcu.table_schema
+              JOIN constraint_column_usage_ AS ccu
+                   ON ccu.constraint_name = tc.constraint_name
+                       AND ccu.table_schema = tc.table_schema
+              JOIN filtered_tables_foreign_keys AS ftfk
+                   ON ftfk.conname = tc.constraint_name
+     WHERE tc.constraint_type = 'FOREIGN KEY'
+       AND tc.constraint_schema = 'ehr'
+       AND kcu.column_name = 'sys_tenant'
+       AND ccu.table_name = 'tenant');
+
+-- create temporary policy
+DO
+$$
+    DECLARE
+        table_name               TEXT;
+        migration_executing_user TEXT := current_user;
+    BEGIN
+        FOR table_name IN SELECT ft.table_name FROM filtered_sys_tenant_tables ft
+            LOOP
+                EXECUTE format('CREATE POLICY ehr_policy_ehrbase_migration ON ehr.%I FOR ALL TO %I USING (TRUE)',
+                               table_name, migration_executing_user);
+            END LOOP;
+    END
+$$;
+
+-- Recreate tenant related foreign keys
+DO
+$$
+    DECLARE
+        FK_records record;
+        query      text;
+    BEGIN
+        RAISE NOTICE '----Start recreating foreign keys-----';
+        FOR FK_records IN (select * from filtered_sys_tenant_tables)
+            LOOP
+                query := 'ALTER TABLE  ehr.' || FK_records.table_name || ' DROP CONSTRAINT ' ||
+                         FK_records.constraint_name || ', ADD CONSTRAINT ' ||
+                         FK_records.constraint_name || ' ' ||
+                         REPLACE(FK_records.constraint_def, ' ON DELETE CASCADE', '') ||
+                         ' ON DELETE CASCADE';
+
+                RAISE NOTICE '%', query;
+                EXECUTE (query);
+            END LOOP;
+        RAISE NOTICE '----Stop recreating foreign keys-----';
+    END
+$$;
+
+-- Drop temporary policy
+DO
+$$
+    DECLARE
```

Comment:
instead of having this super complicated scrip which calculates  the  needed  alter table statements,

way not run this one in you locale and then copy the resulting statements in the scrips.